tools/integration_tests/read_cache/helpers_test.go

```diff
@@ -107,27 +110,40 @@ func getCachedFilePath(fileName string) string {
 }
 
 func validateFileSizeInCacheDirectory(fileName string, filesize int64, t *testing.T) {
-	// Validate that the file is present in cache location.
+	maxRetries := 10
+	retryDelay := 500 * time.Millisecond
 	expectedPathOfCachedFile := getCachedFilePath(fileName)
-	fileInfo, err := operations.StatFile(expectedPathOfCachedFile)
-	if err != nil {
-		t.Errorf("Failed to find cached file %s: %v", expectedPathOfCachedFile, err)
-	}
-	// Validate file size in cache directory matches actual file size.
-	if (*fileInfo).Size() != filesize {
-		t.Errorf("Incorrect cached file size. Expected %d, Got: %d", filesize, (*fileInfo).Size())
+	var err error
+
+	for attempt := 1; attempt <= maxRetries; attempt++ {
+		// Validate that the file is present in cache location.
+		var fileInfo *fs.FileInfo
+		fileInfo, err = operations.StatFile(expectedPathOfCachedFile)
+		if fileInfo == nil {
+			err = fmt.Errorf("received nil FileInfo %s", expectedPathOfCachedFile)
```

Comment:
It should be t.Fatalf. Otherwise it will panic in below statement while accessing `(*fileInfo).Size()`.

Reply:
err == nil before checking `(*fileInfo).Size()` will take care of this right? I don't want the test execution to stop until retries are done so didn't use t.Fatalf .

Reply:
Okay got it. Thanks